src/lib/UserSetting/ValueDefinitionRegistry.php

```diff
@@ -11,33 +11,30 @@
 use Ibexa\Contracts\User\UserSetting\ValueDefinitionGroupInterface;
 use Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface;
 use Ibexa\Core\Base\Exceptions\InvalidArgumentException;
-use Ibexa\User\UserSetting\Group\CustomGroup;
 
 /**
  * @internal
  */
 class ValueDefinitionRegistry
 {
-    /** @var \Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface[] */
+    /** @var array<string, \Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface> */
     protected array $valueDefinitions;
 
-    /** @var \Ibexa\Contracts\User\UserSetting\ValueDefinitionGroupInterface[] */
-    protected $groupedDefinitions;
+    /** @var array<string, \Ibexa\Contracts\User\UserSetting\ValueDefinitionGroupInterface> */
+    protected array $groupedDefinitions;
 
-    public function __construct(array $valueDefinitions = [])
-    {
+    /**
+     * @param array<string, \Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface> $valueDefinitions
+     */
+    public function __construct(
+        array $valueDefinitions = []
+    ) {
         $this->valueDefinitions = [];
         foreach ($valueDefinitions as $identifier => $valueDefinition) {
             $this->valueDefinitions[$identifier] = $valueDefinition;
         }
-        $this->groupedDefinitions[CustomGroup::CUSTOM_GROUP_IDENTIFIER] = $valueDefinitions;
```

Comment:
I'm not sure about this one, `$fieldDefinitions` is an array of `ValueDefinitionInterface`, but `groupedDefinitions` requires array of  `ValueDefinitionGroupInterface`, maybe someone has an idea why do we need it here

Reply:
Seems like it's not needed, groups are properly resolved via compiler pass